### PR TITLE
1.21 Update max money to 10 billion

### PR DESCRIPTION
**Pull request by rnicoll**
Update maximum money to 10 billion coins (most you can have in a single transaction).

### CODE REVIEW

**Comment by rnicoll** [comment]
To apply after #1780

File: src/amount.h [hunk 1]
```diff
@@ -22,7 +22,7 @@ static const CAmount COIN = 100000000;
  * critical; in unusual circumstances like a(nother) overflow bug that allowed
  * for the creation of coins out of thin air modification could lead to a fork.
  * */
-static const CAmount MAX_MONEY = 21000000 * COIN;
+static const CAmount MAX_MONEY = 10000000000 * COIN; // Dogecoin: maximum of 100B coins (given some randomness), max transaction 10,000,000,000
```
Comment (williamboman) [inline src/amount.h:25]:
```suggestion
static const CAmount MAX_MONEY = 10000000000 * COIN; // Dogecoin: maximum of 10B coins (given some randomness), max transaction 10,000,000,000
```

Comment (rnicoll) [reply src/amount.h:25]:
```suggestion
static const CAmount MAX_MONEY = 10000000000 * COIN; // Dogecoin: total supply is not finite, but we constrain maximum single transaction to 10,000,000,000
```
Ah, it's meant to refer to total supply, and I realise the comment is a legacy from Dogecoin 0.4.

**Review by tuhuynh27 — APPROVED** [review]
LGTM

**Comment by chattip-gee** [comment]
LGTM

**Review by l4p4 — DISMISSED** [review]
ok!

**Review by patricklodder — APPROVED** [review]
ACK.

**Review by soljarjb13 — COMMENTED** [review]
V

**Pull request merged**